web/modules/custom/usfs_club_search/src/Plugin/Block/ClubsListBlock.php

```diff
@@ -23,13 +31,190 @@ public function build() {
     $config_file = file_get_contents($config_file_path);
     $yml_contents = Yaml::parse($config_file);
 
+    // Receive search params.
+    $search_parameters = \Drupal::request()->query->all();
+
+    // Load Clubs and Search.
     $clubs_api = \Drupal::state()->get('usfs_clubs');
-    $clubs = [$clubs_api[0], $clubs_api[1], $clubs_api[2]];
+    // $clubs = [$clubs_api[0], $clubs_api[1], $clubs_api[2], $clubs_api[3]];
+    $clubs = $clubs_api;
+    unset($clubs['timestamp']);
+    // Perform search.
+    $search_results = [];
+    foreach ($clubs as $club) {
+      $search_enabled = FALSE;
+      // First confirm with keyword.
+      if ($search_parameters['keyword'] == '') {
+        $search_enabled = TRUE;
+      }
+      elseif (strpos($club['clubName'], $search_parameters['keyword']) !== FALSE) {
+        $search_enabled = TRUE;
+      }
+      if ($search_parameters['keyword'] == '') {
+        $search_enabled = TRUE;
+      }
+      elseif (strpos($club['rinkName'], $search_parameters['keyword']) !== FALSE) {
+        $search_enabled = TRUE;
+      }
+      // Then Confirm with state.
+      if ($search_enabled) {
```

Comment:
Same thing here with something like:

```suggestion
      $search_enabled =
        $search_enabled
        && (
          $search_parameters['state'] == 'Any'
          || !isset($search_parameters['state'])
          || $club['state'] == $search_parameters['state'];
        );
```